### PR TITLE
Renew loans

### PR DESCRIPTION
**Pull request by tripox**

### CODE REVIEW

File: modules/aleph/aleph.module [hunk 1]
```diff
@@ -144,12 +144,15 @@ function aleph_client() {
 /**
  * Return an instance of the AlephPatron by account.
  *
+ * @param object $account
+ *    The global Drupal user object.
```
Comment (danquah) [inline modules/aleph/aleph.module]: You don't actually retrive the object - so the fact that it is global is irelevant - just go with "a Drupal account object" and you're good.

File: modules/aleph/includes/aleph.loan.inc [hunk 2]
```diff
@@ -38,3 +38,33 @@ function aleph_loan_list() {
...
+ * @param object $account
+ *    Drupal user object to renew loan for.
+ * @param array $loan_ids
+ *    The loan IDs.
+ *
+ * @return array Result of renewals.
+ *    Result of renewals.
+ */
+function aleph_loan_renew($account, array $loan_ids) {
+  $patron = aleph_get_patron_by_global_account($account);
```
Comment (danquah) [inline modules/aleph/includes/aleph.loan.inc]: Does seem a bit wird to pass an account object to a function whos name indicates that it is going to be using a global account ...

Comment (danquah) [inline modules/aleph/includes/aleph.loan.inc]: What happens if the patron can't be loaded?

File: modules/aleph/src/Aleph/AlephClient.php [hunk 3]
```diff
@@ -223,13 +238,26 @@ public function getItems(AlephMaterial $material) {
...
+   *
+   * @return \SimpleXMLElement The response from Aleph.
    *    The response from Aleph.
    *
-   * * @throws \RuntimeException
+   * @throws \RuntimeException
    */
-  public function getLoans(AlephPatron $patron) {
-    return $this->requestRest('GET', 'patron/' . $patron->getId() . '/circulationActions/loans?view=full');
+  public function getLoans(AlephPatron $patron, $loan_id = FALSE) {
```
Comment (danquah) [inline modules/aleph/src/Aleph/AlephClient.php:249]: Not 100% specific to this line - but in general we should be carefull to differentiate between safe data that comes from objects we've created (and objects in general) - and potentially dangerous "raw" data that might originate directly from a url-parameter that the user can modify. (think injection attacks)

File: modules/aleph/src/Aleph/Entity/AlephLoan.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ *
+ * Entity with setters/getters for loans in Aleph.
+ *
+ * https://www.obvsg.at/uploads/media/Z36.pdf
+ *
+ * @package Drupal\aleph\Aleph
+ */
+
+class AlephLoan {
+  protected $loanId;
```
Comment (danquah) [inline modules/aleph/src/Aleph/Entity/AlephLoan.php]: Comments and types would be nice here.

**Review by danquah — CHANGES_REQUESTED** [review]

File: modules/aleph/includes/aleph.loan.inc (hunk 2 above)
Comment (tripox) [reply modules/aleph/includes/aleph.loan.inc]: `ding_user_get_creds()`  in `aleph_get_patron_by_account()` throws a `DingProviderAuthException`.

**Review by danquah — APPROVED** [review]

**Pull request merged**